lti_consumer/lti_1p3/consumer.py

```diff
@@ -493,3 +515,101 @@ def enable_ags(
+            preflight_response,
+            resource_link
+    ):
+        """
+        Build LTI message for Deep linking launches.
+
+        Overrides method from LtiConsumer1p3 to allow handling LTI Deep linking messages
+        """
+        # Check if Deep Linking is enabled and that this is a Deep Link Launch
+        if self.dl and preflight_response.get("lti_message_hint") == "deep_linking_launch":
+            # Validate preflight response
+            self._validate_preflight_response(preflight_response)
+
+            # Get LTI Launch Message
+            lti_launch_message = self.get_lti_launch_message(
+                resource_link=resource_link,
+                include_extra_claims=False,
+            )
+
+            # Update message type to LtiDeepLinkingRequest,
+            # replacing the normal launch request.
+            lti_launch_message.update({
+                "https://purl.imsglobal.org/spec/lti/claim/message_type": "LtiDeepLinkingRequest",
+            })
+            # Include deep linking claim
+            lti_launch_message.update(
+                # TODO: Add extra settings
+                self.dl.get_lti_deep_linking_launch_claim()
+            )
+
+            # Nonce from OIDC preflight launch request
+            lti_launch_message.update({
+                "nonce": preflight_response.get("nonce")
+            })
+
+            # Return new lanch message, used by XBlock to present the launch
+            return {
+                "state": preflight_response.get("state"),
+                "id_token": self.key_handler.encode_and_sign(
+                    message=lti_launch_message,
+                    expiration=300
+                )
+            }
+
+        # Call LTI Launch if Deep Linking is not
+        # set up or this isn't a Deep Link Launch
+        return super().generate_launch_request(
+            preflight_response,
+            resource_link
+        )
+
+    def check_and_decode_deep_linking_token(self, token):
+        """
+        Check and decode Deep Linking response, return selected content items.
+
+        This either returns a content item list or raises an exception.
+        """
+        if not self.dl:
+            raise exceptions.LtiAdvantageServiceNotSetUp()
+
+        # Decode token, check expiration
+        deep_link_response = self.tool_jwt.validate_and_decode(token)
+
+        # Check the response is a Deep Linking response type
+        message_type = deep_link_response.get("https://purl.imsglobal.org/spec/lti/claim/message_type")
+        if not message_type == "LtiDeepLinkingResponse":
+            raise exceptions.InvalidClaimValue("Token isn't a Deep Linking Response message.")
+
+        # Check if supported contentitems were returned
+        content_items = deep_link_response.get(
+            'https://purl.imsglobal.org/spec/lti-dl/claim/content_items',
+            # If not found, return empty list
+            [],
+        )
+        if any([
```

Comment:
Python nit: no need for square brackets here.  any() will work with a generator expression.

Reply:
Fixed!